superset/views/database/views.py

```diff
@@ -307,16 +306,29 @@ def form_post(self, form: ExcelToDatabaseForm) -> Response:
             database = (
                 db.session.query(models.Database).filter_by(id=con.data.get("id")).one()
             )
+
+            # some params are not supported by pandas.read_excel (e.g. chunksize).
+            # More can be found here:
+            # https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_excel.html
             excel_to_df_kwargs = {
                 "header": form.header.data if form.header.data else 0,
                 "index_col": form.index_col.data,
                 "mangle_dupe_cols": form.mangle_dupe_cols.data,
-                "skipinitialspace": form.skipinitialspace.data,
                 "skiprows": form.skiprows.data,
                 "nrows": form.nrows.data,
-                "sheet_name": form.sheet_name.data,
-                "chunksize": 1000,
+                "sheet_name": (
+                    int(form.sheet_name.data)
+                    if form.sheet_name.data.isdigit()
+                    else form.sheet_name.data
+                )
+                if form.sheet_name.data
+                else 0,
```

Comment:
While I understand the convenience aspect of supporting both `str` and `int` here, this won't work if a sheet name is truly a digit. I'd prefer to standardize on a single approach, i.e. either only support integer values (=zero indexed position) or assume that any value is the sheet name. Personally I'd vote for the following option:
- if no value is provided, assume it's the first sheet.
- if a value is provided, treat it as a string. Entering the value "1" would look for a sheet with the name "1", not the second sheet.

Reply:
@villebro, I've taken your advice and change the code, thanks.